.github/workflows/tests.yml

```diff
@@ -0,0 +1,22 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint_docs:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v1
+
+    - name: Install dependencies
+      run:  python3 -m pip install -r docs/requirements.txt
+
+    - name: Run files generation tests
+      run:  pre-commit run --all-files && [[ -z "$(git status -s)" ]]
```

Comment:
What's  `&& [[ -z "$(git status -s)" ]]` for?

Reply:
making sure that nothing changed :

* `&&` : run the next command only if the previous succeeded
* `git status -s` : only print the files that differs (including new and deleted)
* `[[ -z ... ]]` : checks if the length of the output or variable is zero and returns true if it is zero.

Reply:
Note to self, we should do `pre-commit run --all-files preprocess-py` to only run this hook in particular.
(I locally added other hooks in my pre-commit config. If we ever add some other hooks for writing the examples (e.g. ruff, black, etc, then we should probably only run the generate hook here)